libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/MultiLanguageGeneratorBase.cs

```diff
@@ -52,7 +53,7 @@ public MultiLanguageGeneratorBase()
         /// <returns>The generator.</returns>
         public override async Task<object> GenerateAsync(DialogContext dialogContext, string template, object data, CancellationToken cancellationToken = default)
         {
-            var targetLocale = dialogContext.Context.Activity.Locale?.ToLowerInvariant() ?? string.Empty;
+            var targetLocale = dialogContext.GetLocale() ?? dialogContext.Context.Activity.Locale ?? string.Empty;
```

Comment:
Ah this one was missing that that is why I didn't see the locale change in my sample. Nice.